### PR TITLE
Added database + few other changes

### PR DESCRIPTION
**Pull request by tensei**
i have to test the systemd service on the server

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -170,38 +169,86 @@ func (b *bot) mute(m dggchat.Message, s *dggchat.Session) {
...
+	}
+}
+
 // !addcommand command response
 func (b *bot) addCommand(m dggchat.Message, s *dggchat.Session) {
 	if !isMod(m.Sender) || !strings.HasPrefix(m.Message, "!addcommand") {
 		return
 	}
 
 	// message itself can contain spaces
```
Comment (xDashh) [inline commands.go:209]: Have to remember why I even added that comment...

File: commands.go [hunk 2]
```diff
@@ -38,10 +32,15 @@ func (b *bot) sendMessageDedupe(m string, s *dggchat.Session) {
...
-	for command, response := range commands {
-		if strings.HasPrefix(m.Message, command) {
+	b.commandMutex.Lock()
+	defer b.commandMutex.Unlock()
 
-			b.sendMessageDedupe(response, s)
+	for _, c := range b.commands {
+		if strings.HasPrefix(m.Message, c.Call) {
+			b.sendMessageDedupe(c.Message, s)
+			c.LastUse = time.Now().UTC()
```
Comment (xDashh) [inline commands.go:41]: Is that for some future feature, like rate-iimiting?

File: commands.go [hunk 3]
```diff
@@ -170,38 +169,86 @@ func (b *bot) mute(m dggchat.Message, s *dggchat.Session) {
...
+	cmnd := strings.TrimSpace(parts[1])
+
+	b.commandMutex.Lock()
+	defer b.commandMutex.Unlock()
+
+	for i, c := range b.commands {
+		if strings.EqualFold(c.Call, cmnd) {
+			err := b.DeleteStaticCommands(c)
+			if err != nil {
+				log.Printf("[##] error deleting command: %v", err)
```
Comment (xDashh) [inline commands.go:192]: `log.Printf`(almost) always needs a `\n` at the end. Some more occurences in this code.

File: main.go [hunk 4]
```diff
@@ -95,19 +76,19 @@ func main() {
 
 	info, err := b.getProfileInfo()
 	if err != nil {
-		debuglogger.Printf("userinfo: %s\n", err.Error())
+		debuglogger.Printf("userinfo: %s", err.Error())
```
Comment (xDashh) [inline main.go:79]: Why remove newlines?

File: etc/systemd/system/modbot.service [hunk 5]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Strims.gg modbot
+After=network.target
+
+[Service]
+User=rustla
+Restart=always
+WorkingDirectory=/home/rustla/modbot
+ExecStart=/usr/bin/modbot
```
Comment (xDashh) [inline etc/systemd/system/modbot.service:9]: Do we (go)install the modbot? probably just use the binary in home path for convenience...

also im not familiar with adding git files with an absolute path, need to check this out. (the web review interface seems to show that)

File: database.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import (
+	"log"
+	"time"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/sqlite"
+)
+
+type staticCommand struct {
+	ID        uint `gorm:"primary_key"`
```
Comment (xDashh) [inline database.go:12]: Can probably use the command name as primary key - is that "Call"?

File: database.go [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+	LastUse   time.Time
+}
+
+func (b *bot) newDatabase() {
+	var err error
+	b.db, err = gorm.Open("sqlite3", b.config.Database)
+	if err != nil {
+		log.Fatalf("failed opening database: %s", err)
+	}
+	b.db.AutoMigrate(&staticCommand{})
```
Comment (xDashh) [inline database.go:29]: gorm has a verbose logging function, probably use that and dumb to the debuglogger or look for gorm errors. Most functions have an `.Error` member iirc.

File: commands.go (hunk 2 above)
Comment (tensei) [reply commands.go:41]: can be or to see if it can be retired after no use for a long time

File: commands.go (hunk 3 above)
Comment (tensei) [reply commands.go:192]: i never needed /n in printf and sometimes my editor tells me to remove it

File: etc/systemd/system/modbot.service (hunk 5 above)
Comment (tensei) [reply etc/systemd/system/modbot.service:9]: yes go install
if you don't want that, we can change the path easily

File: commands.go (hunk 3 above)
Comment (xDashh) [reply commands.go:192]: Figured it out: https://godoc.org/log
"Every log message is output on a separate line: if the message being printed does not end in a newline, the logger will add one."
I'd guess editor telling you to remove something happens in Sprintf commands where the strings might be re-used in another string :thinking:

**Pull request closed without merging**